node/src/main/java/eu/cloudnetservice/node/service/defaults/AbstractService.java

```diff
@@ -498,9 +498,17 @@ protected void executeDeployment(@NonNull ServiceDeployment deployment) {
         var fileName = Files.isDirectory(path)
           ? path.getFileName().toString() + '/'
           : path.getFileName().toString();
-        // check if the file is ignored
-        return deployment.excludes().stream().noneMatch(pattern -> fileName.matches(pattern.replace("*", "(.*)")))
-          && !DEFAULT_DEPLOYMENT_EXCLUSIONS.contains(fileName);
+        // file matcher escaping asterix chars for regex
+        Predicate<String> fileMatcher = input -> fileName.matches(input.replace("*", "(.*)"));
+        // if we have inclusions we just handle them, exclusions otherwise
+        if (deployment.includes().isEmpty()) {
+          // check if the file is excluded
+          return deployment.excludes().stream().noneMatch(fileMatcher)
+            && !DEFAULT_DEPLOYMENT_EXCLUSIONS.contains(fileName);
```

Comment:
Maybe an idea for later as well, but we could move the default exclusions into the service deployment builder and add a method to exclude them 🤔 rather than to enforce that exclusion...